examples/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   cmake_minimum_required(VERSION 3.12)
```

Comment:
```suggestion
  cmake_minimum_required(VERSION 3.16)
```

Reply:
Why?

Reply:
For one you can't build neither ArborX nor Kokkos 3.4 without CMake >=3.16

Reply:
Can a version of CMake that builds code that links against ArborX be lower than the version that ArborX is built with?

Reply:
@aprokop you can but we actually require CMake 3.16 to build ArborX https://github.com/arborx/ArborX/blob/master/CMakeLists.txt#L1. I guess we just forgot to update the version here.